### PR TITLE
Add support for rendering variable in rule objects

### PR DESCRIPTION
**Pull request by Vincent056**
This PR will will render variable in description and rationale for rule objects.

### CODE REVIEW

File: pkg/profileparser/profileparser.go [hunk 1]
```diff
@@ -541,6 +541,23 @@ func ParseRulesAndDo(contentDom *xmlquery.Node, stdParser *referenceParser, pb *
...
 
+	allValues := xmlquery.Find(contentDom, "//xccdf-1.2:Value")
+	valuesList := make(map[string]string)
+	for _, variable := range allValues {
+		for _, val := range variable.SelectElements("//xccdf-1.2:value") {
+			if val.SelectAttr("hidden") == "true" {
+				// this is typically used for functions
+				continue
+			}
+			if val.SelectAttr("selector") == "" {
```
Comment (jhrozek) [inline pkg/profileparser/profileparser.go:552]: Can you remind me what the empty selector means?

File: pkg/profileparser/profileparser.go [hunk 2]
```diff
@@ -621,10 +638,11 @@ func ParseRulesAndDo(contentDom *xmlquery.Node, stdParser *referenceParser, pb *
 				},
 			}
 			if description != nil {
-				p.Description = utils.XmlNodeAsMarkdown(description)
+				p.Description, _ = utils.RenderValues(utils.XmlNodeAsMarkdownPreRender(description), valuesList)
```
Comment (jhrozek) [inline pkg/profileparser/profileparser.go:641]: I agree with not passing an error up, but I think we could at least log the error and not ignore it.

File: pkg/utils/xml2text.go [hunk 3]
```diff
@@ -32,9 +41,14 @@ func xmlToHtml(in string) string {
 			// ignore errors and try to format as much as possible
 			continue
 		}
+
 		switch tok := t.(type) {
 		case xml.StartElement:
-			builder.WriteString(formatElement(tok.Name, "<"))
+			if preRender && tok.Name.Local == "sub" && len(tok.Attr) > 1 {
```
Comment (jhrozek) [inline pkg/utils/xml2text.go:47]: So can the subelement only ever refer to a value or should the substitution be only done if `tok.Attr[0]` has a variable prefix?

File: pkg/profileparser/profileparser.go (hunk 1 above)
Comment (Vincent056) [reply pkg/profileparser/profileparser.go:552]: Not present empty selector means the value is not one of selection values, and it is a default value

**Comment by Vincent056** [comment]
@JAORMX I added the variable used annotations for rules

**Pull request merged**